core/domain/play_later_services.py

```diff
@@ -0,0 +1,239 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Services for the play later feature of the learner dashboard."""
+
+from core.domain import subscription_services
+from core.domain import user_domain
+from core.platform import models
+
+(user_models,) = models.Registry.import_models([models.NAMES.user])
+
+def get_play_later_activities_from_model(play_later_activities_model):
+    """Returns the play later activities domain object given the play later
+    model loaded from the datastore.
+
+    Args:
+        play_later_activities_model: PlayLaterActivitiesModel. The
+            activities completed model loaded from the datastore.
+
+    Returns:
+        PlayLaterActivities. The play later activities corresponding to the
+            given model.
+    """
+    return user_domain.PlayLaterActivities(
+        play_later_activities_model.id,
+        play_later_activities_model.exploration_ids,
+        play_later_activities_model.collection_ids)
+
+
+def save_play_later_activities(play_later_activities):
+    """Save an play later activities domain object as an
+    PlayLaterActivitiesModel entity in the datastore.
+
+    Args:
+        play_later_activities: PlayLaterActivities. The play later
+            activities domain object to be saved in the datastore.
+    """
+    play_later_activities_model = user_models.PlayLaterActivitiesModel(
+        id=play_later_activities.id,
+        exploration_ids=(
+            play_later_activities.exploration_ids),
+        collection_ids=play_later_activities.collection_ids)
+
+    play_later_activities_model.put()
+
+
+def mark_exploration_to_be_played_later(
+        user_id, exploration_id, position_to_be_inserted=None):
+    """Adds the exploration id to the play later list of the user at the given
+    position. If the position is not specified, the exploration gets added at
+    the end. If the exploration is already completed or currently being
+    completed it is not added. The maximum limit of the play later list is 10.
+    If the count exceeds 10, the exploration is not added.
+
+    Args:
+        user_id: str. The id of the user.
+        exploration_id: str. The id of the exploration to be added to the play
+            later list.
+        position_to_be_inserted: int|None. If this is specified the exploration
+            gets inserted at the given position. Otherwise it gets added at the
+            end.
+    """
+    # Need to import learner_progress_services so that we are able to fetch the
```

Comment:
Done! I have added a function in learner_progress_services to check the completed/incomplete parts and then the rest of the functionality passes on to the learner_playlist_services.py. No special tests have been added for the functions in learner_progress_services as their functionality is tested in the tests already added.